kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -98,13 +97,19 @@ def check(self, instance):
 
         cli = self._get_kafka_client(instance)
         cli._maybe_refresh_metadata()
-        kafka_version = self._get_kafka_version(cli)
+
         if get_kafka_consumer_offsets:
             # For now, consumer groups are mandatory if not using ZK
             if not zk_hosts_ports and not consumer_groups:
                 raise BadKafkaConsumerConfiguration('Invalid configuration - if you\'re not collecting '
                                                     'offsets from ZK you _must_ specify consumer groups')
-            if self._kafka_compatible(kafka_version):
+            # kafka-python automatically probes the cluster for broker version
+            # and then stores it. Note that this returns the first version
+            # found, so in a mixed-version cluster this will be a
+            # non-deterministic result.
+            #
+            # Kafka 0.8.2 added support for storing consumer offsets in Kafka.
+            if client.config.get('api_version') >= (0, 8, 2):
```

Comment:
I believe `client` should be `cli`.

Reply:
excellent catch. That's what I get for relying on the unit test harness here I guess (still doing additional refactoring before I start testing with it against my clusters).

@masci any idea why the recent switch to pytest didn't catch this? Is there no test for the `kafka_consumer_offsets: True` scenario?

Reply:
https://gitlab.com/gitlab-org/gitlab-ee/issues/5667